pkgs/cli/src/cli_integration_tests.zig

```diff
@@ -0,0 +1,101 @@
+const std = @import("std");
+const process = std.process;
+const net = std.net;
+
+test "CLI beam command with mock network - complete integration test" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    // Start CLI with beam command and mock network
+    const args = [_][]const u8{ "./zig-out/bin/zeam", "beam", "--mockNetwork", "true" };
```

Comment:
move this to the test direcrectory  in cli package and just rename file to `integration.zig`